client/jetpack-cloud/sections/partner-portal/company-details-form/index.tsx

```diff
@@ -66,10 +70,29 @@ export default function CompanyDetailsForm( {
 	const [ contactPerson, setContactPerson ] = useState( initialValues.contactPerson ?? '' );
 	const [ companyWebsite, setCompanyWebsite ] = useState( initialValues.companyWebsite ?? '' );
 	const [ companyType, setCompanyType ] = useState( initialValues.companyType ?? '' );
+	const [ partnerProgramOptIn, setPartnerProgramOptIn ] = useState( false );
+
+	const [ showPartnerProgramOptIn, setShowPartnerProgramOptIn ] = useState( false );
 
 	const country = getCountry( countryValue, countryOptions );
 	const stateOptions = stateOptionsMap[ country ];
 
+	const handleCompanyTypeChange = ( event: ChangeEvent< HTMLInputElement > ) => {
+		setCompanyType( event.target.value );
+
+		const isEligibleForPartnerProgram = companyTypesEligibleForPartnerProgram.includes(
+			event.target.value
+		);
+		setShowPartnerProgramOptIn( isEligibleForPartnerProgram );
+	};
```

Comment:
Minor suggestion, only because the value is used twice:
```suggestion
	const handleCompanyTypeChange = ( event: ChangeEvent< HTMLInputElement > ) => {
		const selectedType = event.target.value;
		setCompanyType( selectedType );

		const isEligibleForPartnerProgram = companyTypesEligibleForPartnerProgram.includes( selectedType );
		setShowPartnerProgramOptIn( isEligibleForPartnerProgram );
	};
```